src/NuGet.Clients/NuGet.Credentials/SecurePluginCredentialProvider.cs

```diff
@@ -81,24 +82,42 @@ public SecurePluginCredentialProvider(IPluginManager pluginManager, PluginDiscov
 
             var plugin = await _pluginManager.CreateSourceAgnosticPluginAsync(_discoveredPlugin, cancellationToken);
 
+            if (!string.IsNullOrEmpty(plugin.Message))
+            {
+                // There is a potential here for double logging as the CredentialService itself catches the exceptions and tries to log it.
+                // In reality the logger in the Credential Service will be null because the first request always comes from a resource provider (ServiceIndex provider) 
+                _logger.LogError(plugin.Message);
+                throw new PluginException(plugin.Message); // Throwing here will block authentication and ensure that the complete operation fails 
+            }
+
             _isAnAuthenticationPlugin = plugin.Claims.Contains(OperationClaim.Authentication);
 
             if (_isAnAuthenticationPlugin)
             {
                 AddOrUpdateLogger(plugin.Plugin);
+                await SetPluginLogLevelAsync(plugin, _logger, cancellationToken);
 
                 if (proxy != null)
                 {
                     await SetProxyCredentialsToPlugin(uri, proxy, plugin, cancellationToken);
                 }
 
                 var request = new GetAuthenticationCredentialsRequest(uri, isRetry, nonInteractive);
-
                 var credentialResponse = await plugin.Plugin.Connection.SendRequestAndReceiveResponseAsync<GetAuthenticationCredentialsRequest, GetAuthenticationCredentialsResponse>(
                     MessageMethod.GetAuthenticationCredentials,
                     request,
                     cancellationToken);
+                if (credentialResponse.ResponseCode == MessageResponseCode.NotFound && nonInteractive)
+                {
+                    _logger.LogWarning(
+                        string.Format(
+                                CultureInfo.CurrentCulture,
+                                Resources.SecurePluginWarning_UseInteractiveOption)
+                                );
```

Comment:
you dont need a new line for `);`

Reply:
It's more readable if you don't have multiple `)`s together :)